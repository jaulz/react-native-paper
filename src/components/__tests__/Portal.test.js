/* @flow */

import * as React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import Portal from '../Portal/Portal';

it('renders portal with siblings', () => {
  const tree = renderer
    .create(
      <Portal.Host>
        <Text>Outside content</Text>
        <Portal>
          <Text>Portal content</Text>
        </Portal>
      </Portal.Host>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
