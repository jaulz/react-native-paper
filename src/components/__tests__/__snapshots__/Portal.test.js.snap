// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders portal with siblings 1`] = `
Array [
  <View
    collapsable={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Outside content
    </Text>
  </View>,
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Portal content
    </Text>
  </View>,
]
`;
